src/NUnitFramework/framework/TestContext.cs

```diff
@@ -103,6 +104,17 @@ public string WorkDirectory
             }
         }
 
+        public static CompatibilityLogger Compatibility
+        {
+            get
+            {
+                var contextDictionary = CurrentContext._context;
+                return contextDictionary != null
+                    ? new CompatibilityLogger((TextWriter)contextDictionary["CompatibilityWriter"])
+                    : null;
```

Comment:
Ah, I see. `CompatibilityLogger` acts as a null object when given a null TextWriter.
When might contextDictionary be null and why wouldn't it be better to impleemnt this getter as:

```cs
return new CompatibilityLogger((TextWriter)CurrentContext._context?["CompatibilityWriter"])
```

Reply:
> The null situation can/should only occur if our code tries to call CompatibilityLogger from outside the execution of a test.

The point not being about syntax per se, but rather, wouldn't it be more consistent with the Compatibility class to return `new CompatibilityLogger(null)` rather than `null`?

Reply:
Gotcha! You're right, that's a better pattern.

Reply:
To the extent that it matters, I would lean towards `(TextWriter)contextDictionary["CompatibilityWriter"]` because if that key is not `TestWriter` and not `null`, I'd want it to stop and say something is wrong. But I highly doubt this will be an issue in practice.